static/src/javascripts/projects/commercial/modules/dfp/apply-creative-template.js

```diff
@@ -0,0 +1,132 @@
+import scrollable from 'commercial/modules/creatives/scrollable-mpu-v2';
+
+const creativeLookup: Object = {
+    frame,
+    revealer,
+    'fabric-v1': fabric,
+    'fabric-expanding-v1': fabricExpand,
+    'fabric-expandable-video-v2': fabricExpandVideo,
+    'fabric-video': fabricVideo,
+    'scrollable-mpu-v2': scrollable,
+};
+
+const renderCreativeTemplate = (
+    adSlot: HTMLElement,
+    iFrame: HTMLIFrameElement
+): Promise<boolean> => {
+    const fetchCreativeConfig = (): ?string => {
+        try {
+            const breakoutScript = iFrame.contentDocument.body
+                ? iFrame.contentDocument.body.querySelector(
+                      '.breakout__script[type="application/json"]'
+                  )
+                : null;
+            return breakoutScript ? breakoutScript.innerHTML : null;
+        } catch (err) {
+            return null;
+        }
+    };
+
+    const mergeViewabilityTracker = (json: Object): Object => {
+        const viewabilityTrackerDiv = iFrame.contentDocument.getElementById(
+            'viewabilityTracker'
+        );
+        const viewabilityTracker = viewabilityTrackerDiv
+            ? viewabilityTrackerDiv.childNodes[0].nodeValue.trim()
+            : null;
+
+        /* eslint-disable no-param-reassign */
+        if (viewabilityTracker && json.params) {
+            json.params.viewabilityTracker = viewabilityTracker;
+        }
+        /* eslint-enable no-param-reassign */
+
+        return json;
+    };
+
+    const renderCreative = (config: Object): Promise<boolean> =>
+        new Promise(resolve => {
+            const Creative = creativeLookup[config.name];
+            resolve(new Creative(adSlot, config.params, config.opts).create());
+        });
+
+    const hideIframe = (): Promise<any> =>
+        fastdom.write(() => {
+            /* eslint-disable no-param-reassign */
+            iFrame.style.display = 'none';
+            /* eslint-enable no-param-reassign */
+        });
+
+    const creativeConfig: ?string = fetchCreativeConfig();
+
+    if (creativeConfig) {
+        return hideIframe()
+            .then(() => JSON.parse(creativeConfig))
+            .then(mergeViewabilityTracker)
+            .then(renderCreative)
+            .catch(err =>
+                reportError(
+                    Error(`Failed to get creative JSON ${err}`),
+                    { feature: 'commercial' },
+                    false
+                )
+            );
+    }
+    return Promise.resolve(true);
+};
+
+const getAdvertIframe = (adSlot: HTMLElement): Promise<HTMLIFrameElement> =>
+    new Promise((resolve, reject) => {
+        // DFP will sometimes return empty iframes, denoted with a '__hidden__' parameter embedded in its ID.
+        // We need to be sure only to select the ad content frame.
+        const contentFrame = ((adSlot.querySelector(
+            'iframe:not([id*="__hidden__"])'
+        ): any): ?HTMLIFrameElement);
```

Comment:
I don't think I can use `instanceof` as this slightly odd type declaration is just to satisfy flow. Unless I'm mistaking what you are suggesting...?